infra/serverless/lambda-web-app-test/main.tf

```diff
@@ -0,0 +1,220 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+#Lambda IAM Role
+resource "aws_iam_role" "iam_role_for_lambda" {
+  name = "iam_role_for_lambda"
+
+  assume_role_policy = <<EOS
+    {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+            "Service": "lambda.amazonaws.com"
+        },
+        "Effect": "Allow",
+        "Sid": ""
+        }
+      ]
+    }
+  EOS
+}
+
+#Lambda Functon Setup
+resource "aws_lambda_function" "weather_app_lambda" {

```

Comment:
When setting up resource the first bracket is what resource and second is what you're naming it?